app/models/database_bloat.rb

```diff
@@ -0,0 +1,505 @@
+        db.send(meth)
+      end
+    end
+  end
+
+  def show_cache_hits!
+    cache_hit_rates.each do |row|
+      Rails.logger.warn "Cache hit rate: #{row}"
+    end
+  end
+
+  def index_drops!
+    unused_indexes.each do |row|
+      Rails.logger.info("EVIDENCE: #{row}")
+      run("DROP INDEX \"#{row['index']}\";")
+    end
+  end
+
+  # Runs in the background and doesn't get an aggresive lock on anything.
+  # Should be safe unless we're low on I/O burst credits
+  def reindex!
+    catch(:enough) do
+      bloated_indexes.each.with_index do |row, i|
+        sql = %<REINDEX INDEX CONCURRENTLY "#{row['schemaname']}"."#{row['idxname']}";>
+        Rails.logger.info("EVIDENCE: #{row}")
+        run(sql)
+        throw :enough if i + 1 == MAX_PER_RUN
+      end
+    end
+  end
+
+  # This aquires an aggresive (exclusive) lock.
+  def vacuum_full!
+    catch(:enough) do
+      bloated_tables.each.with_index do |row, i|
+        sql = %<VACUUM FULL "#{row['schemaname']}"."#{row['tblname']}";>
+        Rails.logger.info("EVIDENCE: #{row}")
+
+        run(sql)
+
+        adjust_autovacuum_for(row)
+
+        throw :enough if i + 1 == MAX_PER_RUN
+      end
+    end
+  end
+
+  def adjust_autovacuum_for(row)
+    scale_factor = [row['autovacuum_vacuum_scale_factor'] * 0.75, 0.005].min
+    sql = %<ALTER TABLE "#{row['schemaname']}"."#{row['tblname']}" SET (autovacuum_analyze_scale_factor = #{scale_factor});>
+    run(sql)
+  end
+
+  # This is like a vacuum full, but orchastrated by pg_repack which doesn't
+  # aquire exlusive locks on the table
+  def repack!
+    catch(:enough) do
+      port = ar_base_class.connection_db_config.configuration_hash[:port].presence || "5432"
+      host = File.exist?(ar_base_class.connection_db_config.configuration_hash[:host].to_s) ? CGI.escape(ar_base_class.connection_db_config.configuration_hash[:host]) : ar_base_class.connection_db_config.configuration_hash[:host]
+      username = ar_base_class.connection_db_config.configuration_hash[:username]
+      database = ar_base_class.connection_db_config.configuration_hash[:database]
+      # password should be in your .pgpass file
+
+
+      bloated_tables.each.with_index do |row, i|
+        # puts row
+
+        # need pg_repack to match match database extension. That's why we have
+        # the containerized pg_repack image in our deployed environments
+        options = "--no-superuser-check -U #{username} -d #{database} -h #{host} -p #{port} -t #{row['schemaname']}.#{row['tblname']}"
+
+        cmd = "pg_repack #{options}"
+
+        raise "version of pg_repack needs to match that in the database"
+
+        Rails.logger.info("Repacking #{row['tblname']}")
+        system(cmd)
+
+        if $?.exitstatus != 0
+          raise "running repack failed."
+        end
+
+        adjust_autovacuum_for(row)
+
+        throw :enough if i + 1 == MAX_PER_RUN
+      end
+    end
+  end
+
+  private
+
+  def build_pg_repack(version='1.4.5')
```

Comment:
This doesn't match `SERVER_PG_REPACK_VERSION = '1.4.7'.freeze`

Reply:
Just haven't finished working through repack in the context of all the different versions in open path.